src/peft/tuners/lora/model.py

```diff
@@ -318,8 +317,42 @@ def _create_new_module(lora_config, adapter_name, target, **kwargs):
                     f"Target module {target} is not supported. Currently, only the following modules are supported: "
                     "`torch.nn.Linear`, `torch.nn.Embedding`, `torch.nn.Conv2d`, `transformers.pytorch_utils.Conv1D`."
                 )
-            new_module = Linear(adapter_name, in_features, out_features, bias=bias, **kwargs)
-
+            kwargs["num_bits"] = lora_config.loftq_bits
+            kwargs["num_iter"] = lora_config.loftq_iter
+            if lora_config.loftq_fake:
+                new_module = Linear(
+                    adapter_name,
+                    in_features,
+                    out_features,
+                    weight_copy=target.weight if lora_config.init_lora_weights == "loftq" else None,
+                    bias=bias,
+                    **kwargs,
+                )
+            elif loaded_in_8bit or lora_config.loftq_bits == 8:
```

Comment:
lines 261 and 272 would run when loaded in 8 bit and 4 bit, respectively. This and the below logic would never run if the model is already loaded in 8/4 bits.

Reply:
Could you please elaborate on that? What are the constraints? I think it would be much more intuitive if we require the base model to be quantized when users want to apply LoftQ initialization.